TP ESPECIAL FUNDAMENTOS DE LA CIENCIA DE DATOS 2024 

DATASET: CALIDAD DE AGUA DEL RÍO DE LA PLATA - 2022

INTEGRANTES:
Abril Valentina Valentina Juarez, Matias Müller Gonzales y Julián Elias Rivero

In [140]:
%pip install matplotlib
%pip install fancyimpute

Note: you may need to restart the kernel to use updated packages.



[notice] A new release of pip is available: 24.0 -> 24.3.1
[notice] To update, run: python.exe -m pip install --upgrade pip


Note: you may need to restart the kernel to use updated packages.



[notice] A new release of pip is available: 24.0 -> 24.3.1
[notice] To update, run: python.exe -m pip install --upgrade pip


In [141]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from fancyimpute import IterativeImputer,KNN
from sklearn.impute import KNNImputer

In [142]:
raw_dataset = pd.read_csv("Calidad_de_agua_2022.csv", sep =";")

Desplegamos el dataset para realizar un vistazo generico

In [143]:
raw_dataset

orden                                 sitios codigo       fecha   año  \
0        1           Canal Villanueva y Río Luján  TI001   23/2/2022  2022   
1        2           Río Lujan y Arroyo Caraguatá  TI006   23/2/2022  2022   
2        3            Canal Aliviador y Río Lujan  TI002   23/2/2022  2022   
3        4  Río Carapachay y Arroyo Gallo Fiambre  TI003   23/2/2022  2022   
4        5            Río Reconquista y Río Lujan  TI004   23/2/2022  2022   
..     ...                                    ...    ...         ...   ...   
163     38                  Balneario Palo Blanco  BS092  31/10/2022  2022   
164     39          Diagonal 66 (descarga cloaca)  BS095  31/10/2022  2022   
165     40                     Playa La Bagliardi  BS091  31/10/2022  2022   
166     41                    Balneario Municipal  BS094  31/10/2022  2022   
167     42                      Playa La Balandra  BS093  31/10/2022  2022   

       campaña tem_agua tem_aire           od           ph  ...     dbo_mg_l  \
0       Verano     24.5     23.3          5.3         6.56  ...          6.2   
1       Verano     25.4     23.3         2.25         6.56  ...          5.8   
2       Verano     24.6     23.3         2.94         6.59  ...          1.9   
3       Verano     25.2     23.3         2.22         7.45  ...          5.8   
4       Verano     24.1       20         1.02         6.39  ...          2.6   
..         ...      ...      ...          ...          ...  ...          ...   
163  Primavera       10       12  no se midió  no se midió  ...  no se midió   
164  Primavera       10       12  no se midió  no se midió  ...  no se midió   
165  Primavera       10       12  no se midió  no se midió  ...  no se midió   
166  Primavera       10       12  no se midió  no se midió  ...  no se midió   
167  Primavera       10       12  no se midió  no se midió  ...  no se midió   

    dqo_mg_l turbiedad_ntu hidr_deriv_petr_ug_l cr_total_mg_l cd_total_mg_l  \
0         29            90                <0.10        <0.005        <0.001   
1         29            34                <0.10        <0.005        <0.001   
2         29            17                <0.10        <0.005        <0.001   
3         29            23                <0.10        <0.005        <0.001   
4         59            18                <0.10        <0.005        <0.001   
..       ...           ...                  ...           ...           ...   
163      <30            23                <0.10        <0.005        <0.001   
164       31          18.2                <0.10        <0.005        <0.001   
165      <30            40                <0.10        <0.005        <0.001   
166       39            90                <0.10             5        <0.001   
167      <30            70                <0.10             5        <0.001   

    clorofila_a_ug_l microcistina_ug_l ica             calidad_de_agua  
0                <10             <0.20  55             Muy deteriorada  
1                <10             <0.20  42  Extremadamente deteriorada  
2                <10               0.2  45             Muy deteriorada  
3                <10             <0.20  46             Muy deteriorada  
4                <10             <0.20  44  Extremadamente deteriorada  
..               ...               ...  ..                         ...  
163              2.1             <0.15  43  Extremadamente deteriorada  
164             20.2             <0.15  37  Extremadamente deteriorada  
165              0.2             <0.15  49             Muy deteriorada  
166             10.5             <0.15  39  Extremadamente deteriorada  
167             48.0             <0.15  34  Extremadamente deteriorada  

[168 rows x 31 columns]

Ahora le pedimos un poco de informacion para analizar contexto de los datos

In [144]:
raw_dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 168 entries, 0 to 167
Data columns (total 31 columns):
 #   Column                   Non-Null Count  Dtype 
---  ------                   --------------  ----- 
 0   orden                    168 non-null    int64 
 1   sitios                   168 non-null    object
 2   codigo                   168 non-null    object
 3   fecha                    168 non-null    object
 4   año                      168 non-null    object
 5   campaña                  168 non-null    object
 6   tem_agua                 168 non-null    object
 7   tem_aire                 166 non-null    object
 8   od                       168 non-null    object
 9   ph                       168 non-null    object
 10  olores                   168 non-null    object
 11  color                    168 non-null    object
 12  espumas                  168 non-null    object
 13  mat_susp                 168 non-null    object
 14  colif_fecales_ufc_100ml  167 non-null    o

En total tenemos 31 columnas (atributos) y 168 filas (observaciones). Se puede ver que hay ciertas columnas con NULL y que el tipo de datos practicamente es OBJECT exceptuando la primer columna "orden" que es de tipo INT64.

In [145]:
raw_dataset.head()

orden                                 sitios codigo      fecha   año  \
0      1           Canal Villanueva y Río Luján  TI001  23/2/2022  2022   
1      2           Río Lujan y Arroyo Caraguatá  TI006  23/2/2022  2022   
2      3            Canal Aliviador y Río Lujan  TI002  23/2/2022  2022   
3      4  Río Carapachay y Arroyo Gallo Fiambre  TI003  23/2/2022  2022   
4      5            Río Reconquista y Río Lujan  TI004  23/2/2022  2022   

  campaña tem_agua tem_aire    od    ph  ... dbo_mg_l dqo_mg_l turbiedad_ntu  \
0  Verano     24.5     23.3   5.3  6.56  ...      6.2       29            90   
1  Verano     25.4     23.3  2.25  6.56  ...      5.8       29            34   
2  Verano     24.6     23.3  2.94  6.59  ...      1.9       29            17   
3  Verano     25.2     23.3  2.22  7.45  ...      5.8       29            23   
4  Verano     24.1       20  1.02  6.39  ...      2.6       59            18   

  hidr_deriv_petr_ug_l cr_total_mg_l cd_total_mg_l clorofila_a_ug_l  \
0                <0.10        <0.005        <0.001              <10   
1                <0.10        <0.005        <0.001              <10   
2                <0.10        <0.005        <0.001              <10   
3                <0.10        <0.005        <0.001              <10   
4                <0.10        <0.005        <0.001              <10   

  microcistina_ug_l ica             calidad_de_agua  
0             <0.20  55             Muy deteriorada  
1             <0.20  42  Extremadamente deteriorada  
2               0.2  45             Muy deteriorada  
3             <0.20  46             Muy deteriorada  
4             <0.20  44  Extremadamente deteriorada  

[5 rows x 31 columns]

A simple vista veo que tendremos que tratar lo siguiente:
1) Hay valores que enves de tener un tipo de dato cuantitativo en el atributo, tienen un string categorico que dice "no se midio" o variantes del mismo, por ejemplo en OD(oxigeno disuelto), esto deberia de ser reemplazado NA.
2) Hay datos cuantitativos que adelante del valor numerico tienen un string "<", como en el caso de las mediciones de los contaminantes en el agua.
Podriamos generar un mapa que descriva las observaciones de dicha variable. 
3) Hay datos cualitativos nominales que indican "presencia" o "ausencia" de algún compuesto en el agua, estás variables son dicotomicas, seran reemplazados por {0, 1} de acuerdo a su impacto en la muestra. 
4) Hay datos cualitativos ordinales como "muy deteriorada" o "extremadamente deteriorada" que hacen referencia al agua de esa zona, seran reemplazados mediante transformacion "ordinal encoding" ya que el orden es significativo.

Veamos cuantos valores repetidos tiene el dataset

In [146]:
raw_dataset.duplicated().sum()

np.int64(0)

Por lo visto no existe valores repetidos a simple vista.

Ya que empezaremos a modificar el dataset, una buena practica es copiar el dataset

In [147]:
copy_dataset = raw_dataset.copy()

podemos eliminar columnas que no tienen mucho sentido, como por ejemplo "orden" y vamos a eliminar "año", ya que todas las observaciones se hicieron en 2022 y no agregan mucha informacion

In [148]:
copy_dataset = copy_dataset.drop("orden",axis=1)
copy_dataset = copy_dataset.drop("año",axis=1)

Ahora vamos a ver que columnas tienen nulos

In [149]:
copy_dataset.isna().sum()

sitios                      0
codigo                      0
fecha                       0
campaña                     0
tem_agua                    0
tem_aire                    2
od                          0
ph                          0
olores                      0
color                       0
espumas                     0
mat_susp                    0
colif_fecales_ufc_100ml     1
escher_coli_ufc_100ml       1
enteroc_ufc_100ml           1
nitrato_mg_l                0
nh4_mg_l                    0
p_total_l_mg_l              0
fosf_ortofos_mg_l           0
dbo_mg_l                    1
dqo_mg_l                    0
turbiedad_ntu               0
hidr_deriv_petr_ug_l        0
cr_total_mg_l               0
cd_total_mg_l               0
clorofila_a_ug_l            1
microcistina_ug_l           0
ica                        13
calidad_de_agua            14
dtype: int64

Pareciera que existen muy pocos nulos, pero vamos a ver que sucede con el "no se midió" , asi que vamos a mostrar algunos ejemplos

In [150]:
copy_dataset[copy_dataset["dbo_mg_l"] == "no se midió"].head()

sitios codigo      fecha campaña     tem_agua  \
10         Espigón La Farola  SI021  23/2/2022  Verano  no se midió   
12  Playa Espigón de Pacheco  SI024  23/2/2022  Verano         26.5   
17       Parque de los Niños  CA041  23/2/2022  Verano         26.1   
27             Náutico 400 m  QU062  23/2/2022  Verano           24   
30  Puerto Trinidad calle 47  BZ077  23/2/2022  Verano  no se midió   

       tem_aire           od           ph       olores        color  ...  \
10  no se midió  no se midió  no se midió  no se midió  no se midió  ...   
12           27          5.3         7.49     Ausencia     Ausencia  ...   
17  no se midió          4.8          7.4     Ausencia     Ausencia  ...   
27           23         9.59         7.94     Ausencia     Ausencia  ...   
30  no se midió  no se midió  no se midió  no se midió  no se midió  ...   

       dbo_mg_l     dqo_mg_l turbiedad_ntu hidr_deriv_petr_ug_l cr_total_mg_l  \
10  no se midió  no se midió   no se midió          no se midió   no se midió   
12  no se midió  no se midió   no se midió          no se midió   no se midió   
17  no se midió           29            22          no se midió        <0.005   
27  no se midió           29            26                <0.10        <0.005   
30  no se midió  no se midió   no se midió          no se midió   no se midió   

   cd_total_mg_l clorofila_a_ug_l microcistina_ug_l  ica  calidad_de_agua  
10   no se midió      no se midió       no se midió  NaN              NaN  
12   no se midió      no se midió       no se midió  NaN              NaN  
17        <0.002      no se midió       no se midió   54  Muy deteriorada  
27        <0.001               28                 1   48  Muy deteriorada  
30   no se midió      no se midió       no se midió  NaN              NaN  

[5 rows x 29 columns]

Como pudimos ver hay muchisimos "no se midió" que representa NA, entonces vamos a convertir para todas las columnas
el "no se midió" y las distintas variantes a NA

In [151]:
lista_columnas = copy_dataset.columns

for col in lista_columnas:
    copy_dataset[col] = copy_dataset[col].replace(['no midieron este día', 'no se midió', 'no se determinó','No se midió',
                                                   'no midio la sonda','NA'], pd.NA)

Una vez pasado todos los valores a NA vamos a contar cuantos NA tenemos por cada variable

In [152]:
copy_dataset.isna().sum()

sitios                      0
codigo                      0
fecha                       5
campaña                     5
tem_agua                   23
tem_aire                   24
od                         36
ph                         28
olores                     13
color                      13
espumas                    13
mat_susp                   13
colif_fecales_ufc_100ml    15
escher_coli_ufc_100ml      15
enteroc_ufc_100ml          15
nitrato_mg_l               14
nh4_mg_l                   14
p_total_l_mg_l             21
fosf_ortofos_mg_l          14
dbo_mg_l                   58
dqo_mg_l                   14
turbiedad_ntu              14
hidr_deriv_petr_ug_l       18
cr_total_mg_l              17
cd_total_mg_l              17
clorofila_a_ug_l           18
microcistina_ug_l          18
ica                        14
calidad_de_agua            14
dtype: int64

In [153]:
copy_dataset.isna().sum() / copy_dataset.shape[0]

sitios                    0.000
codigo                    0.000
fecha                     0.030
campaña                   0.030
tem_agua                  0.137
tem_aire                  0.143
od                        0.214
ph                        0.167
olores                    0.077
color                     0.077
espumas                   0.077
mat_susp                  0.077
colif_fecales_ufc_100ml   0.089
escher_coli_ufc_100ml     0.089
enteroc_ufc_100ml         0.089
nitrato_mg_l              0.083
nh4_mg_l                  0.083
p_total_l_mg_l            0.125
fosf_ortofos_mg_l         0.083
dbo_mg_l                  0.345
dqo_mg_l                  0.083
turbiedad_ntu             0.083
hidr_deriv_petr_ug_l      0.107
cr_total_mg_l             0.101
cd_total_mg_l             0.101
clorofila_a_ug_l          0.107
microcistina_ug_l         0.107
ica                       0.083
calidad_de_agua           0.083
dtype: float64

Además vemos que ninguna variable tiene más del 50% de valores faltantes, por ende no eliminaremos ninguna variable 

Se eliminaran todas aquellas observaciones que contengan 14 o mas atributos con valores NA, ya que no nos aportara mucha informacion

In [154]:
copy_dataset = copy_dataset.dropna(thresh=14)

A continuacion estamos mostrando aquellas observaciones donde en la columna microcistina_ug_l se encuentran valores que deberian ser numericos, pero son de tipo object, porque hay texto combinado con numeros. Observamos que hay esto sucede en reiteradas ocasiones.

In [155]:
copy_dataset[copy_dataset["microcistina_ug_l"] == " <0.15"].value_counts()

sitios                                 codigo  fecha      campaña   tem_agua  tem_aire  od     ph    olores     color      espumas   mat_susp   colif_fecales_ufc_100ml  escher_coli_ufc_100ml  enteroc_ufc_100ml  nitrato_mg_l  nh4_mg_l  p_total_l_mg_l  fosf_ortofos_mg_l  dbo_mg_l  dqo_mg_l  turbiedad_ntu  hidr_deriv_petr_ug_l  cr_total_mg_l  cd_total_mg_l  clorofila_a_ug_l  microcistina_ug_l  ica  calidad_de_agua           
Arroyo Abra Vieja y Santa Rosa         TI009   23/8/2022  invierno  15.6      15        8.68   7.13  Ausencia   Ausencia   Ausencia  Ausencia   3200                     2                      250                2.4           0.11      0.12            0.12               <2.0      <30       33              <0.10                <0.005         <0.001         200                <0.15             51   Muy deteriorada               1
Arroyo El Gato                         EN083   23/8/2022  invierno  12.8      9         2.49   8.01  Ausencia   Ausencia   Ausencia  Ausencia  

Para las siguientes columnas fue necesario quitar el string "<" y " <"(este caracter es el denominado NBSP)

In [156]:
columnas = [
    "nh4_mg_l",
    "p_total_l_mg_l",
    "fosf_ortofos_mg_l",
    "dbo_mg_l",
    "dqo_mg_l",
    "turbiedad_ntu",
    "hidr_deriv_petr_ug_l",
    "cr_total_mg_l",
    "cd_total_mg_l",
    "clorofila_a_ug_l",
    "microcistina_ug_l"
]
for col in columnas:
    copy_dataset[col] = copy_dataset[col].str.strip("<")
    copy_dataset[col] = copy_dataset[col].str.strip(" <")

Ahora veamos la variable "cd_total_mg_l" y notaremos que a pesar de quitar el "<" el tipo de dato no cambio, sigue siendo de tipo object

In [157]:
copy_dataset[copy_dataset["cd_total_mg_l"] == "0.001"].value_counts()

sitios                          codigo  fecha      campaña   tem_agua  tem_aire  od    ph    olores     color     espumas   mat_susp   colif_fecales_ufc_100ml  escher_coli_ufc_100ml  enteroc_ufc_100ml  nitrato_mg_l  nh4_mg_l  p_total_l_mg_l  fosf_ortofos_mg_l  dbo_mg_l  dqo_mg_l  turbiedad_ntu  hidr_deriv_petr_ug_l  cr_total_mg_l  cd_total_mg_l  clorofila_a_ug_l  microcistina_ug_l  ica  calidad_de_agua           
Arroyo Abra Vieja y Santa Rosa  TI009   23/2/2022  Verano    23.4      21        4.49  6.76  Ausencia   Ausencia  Ausencia  Ausencia   400                      100                    220                1.9           0.1       0.21            0.19               1.9       29        39             0.10                  0.005          0.001          10                0.20               58   Muy deteriorada               1
                                        23/8/2022  invierno  15.6      15        8.68  7.13  Ausencia   Ausencia  Ausencia  Ausencia   3200                     2

Entonces para esta variable en particular vamos a realizar un mapeo para sustituir el tipo string a tipo entero

In [158]:
mgl_map = {
    "0.001":0.001,
    "0.002":0.002
}
copy_dataset["cd_total_mg_l"] = copy_dataset["cd_total_mg_l"].map(mgl_map)
copy_dataset["cd_total_mg_l"].info()

<class 'pandas.core.series.Series'>
Index: 154 entries, 0 to 167
Series name: cd_total_mg_l
Non-Null Count  Dtype  
--------------  -----  
151 non-null    float64
dtypes: float64(1)
memory usage: 2.4 KB


Para la variable calidad_de_agua hicimos una transformacion ordinal encoding, donde cada valor unico de una caracteristica se mapea a un entero, tratando de ser lo mas cuidadoso para preservar el orden natural

In [159]:
niveles_deterioro = {
    "Deteriorada" : 1,
    "Muy deteriorada" : 2,
    "Extremadamente deteriorada" : 3
}

copy_dataset["calidad_de_agua"] = copy_dataset["calidad_de_agua"].map(niveles_deterioro)

Para las siguientes variables continuaremos con un tipo de mapeo dicotomico

In [160]:
mapeo_dicotomico = {
    "Ausencia" : 0,
    "Ausente" : 0,
    "ausencia" : 0,
    "presencia" : 1,
    "Presencia" : 1
}

copy_dataset["olores"] = copy_dataset["olores"].map(mapeo_dicotomico)
copy_dataset["color"] = copy_dataset["color"].map(mapeo_dicotomico)
copy_dataset["espumas"] = copy_dataset["espumas"].map(mapeo_dicotomico)
copy_dataset["mat_susp"] = copy_dataset["mat_susp"].map(mapeo_dicotomico)

A continuacion vamos a cambiar dos valores de la variable "fecha", ya que la fecha de la medicion se cargo como "31/10/0202" cuando en realidad intumos que se quiso cargar "31/10/2022" debido a que todas las muestras fueron realizadas en dicho año

In [161]:
copy_dataset["fecha"] = copy_dataset["fecha"].replace("31/10/0202", "31/10/2022")
copy_dataset["fecha"].value_counts()

fecha
4/5/2022      41
23/2/2022     39
23/8/2022     37
31/10/2022    37
Name: count, dtype: int64

Para tratar los NA vamos a operar con todas las columnas que lo posean, a traves de una serie de pasos
1_Convertir todas las variables a tipo numerico, exceptuando los NA(aquellos que no puedan convertirse, como los NA, seran convertidos a NAN)
2_Calcularemos la mediana para cada estacion del año
3_Con la funcion fillna, Asignaremos la mediana de cada estacion a los valores NAN en funcion de la columna a la que pertenezca

Decidimos hacer imputacion por mediana y por estacion, para mantener la distribucion de los datos lo mas equilibrada posible.

In [162]:
columnas = [
    "tem_agua",
    "tem_aire",
    "od",
    "ph",
    "colif_fecales_ufc_100ml",
    "escher_coli_ufc_100ml",
    "enteroc_ufc_100ml",
    "nitrato_mg_l",
    "nh4_mg_l",
    "p_total_l_mg_l",
    "fosf_ortofos_mg_l",
    "dbo_mg_l",
    "dqo_mg_l",
    "turbiedad_ntu",
    "hidr_deriv_petr_ug_l",
    "cr_total_mg_l",
    "cd_total_mg_l",
    "clorofila_a_ug_l",
    "microcistina_ug_l",
    "ica"
]
for col in columnas:
    copy_dataset[col] = pd.to_numeric(copy_dataset[col], errors= 'coerce')
    
    median_verano = copy_dataset.loc[copy_dataset["campaña"] == "Verano", col].median()
    median_invierno = copy_dataset.loc[copy_dataset["campaña"] == "invierno", col].median()
    median_otoño = copy_dataset.loc[copy_dataset["campaña"] == "otoño", col].median()
    median_primavera = copy_dataset.loc[copy_dataset["campaña"] == "Primavera", col].median()
    
    copy_dataset.loc[(copy_dataset['campaña'] == 'Verano') & (copy_dataset[col].isna()), col] = median_verano
    copy_dataset.loc[(copy_dataset['campaña'] == 'invierno') & (copy_dataset[col].isna()), col] = median_invierno
    copy_dataset.loc[(copy_dataset['campaña'] == 'Primavera') & (copy_dataset[col].isna()), col] = median_primavera
    copy_dataset.loc[(copy_dataset['campaña'] == 'otoño') & (copy_dataset[col].isna()), col]=median_otoño

c:\Users\Matias\Desktop\TP ESPECIAL - CIENCIA DE DATOS\practicoespecialvenv\Lib\site-packages\numpy\lib\_nanfunctions_impl.py:1241: RuntimeWarning: Mean of empty slice
  return np.nanmean(a, axis, out=out, keepdims=keepdims)


Veamos que pasa con la columna "dbo_mg_l". Esta presenta inconvenientes para transformar el tipo de dato (siguen habiendo Nan)

In [163]:
primavera_nan = copy_dataset[(copy_dataset['campaña'] == 'Primavera') & (copy_dataset['dbo_mg_l'].isna())]
print(primavera_nan)

                                    sitios    codigo       fecha    campaña  \
126           Canal Villanueva y Río Luján     TI001  31/10/2022  Primavera   
127           Río Lujan y Arroyo Caraguatá     TI006  31/10/2022  Primavera   
128            Canal Aliviador y Río Lujan     TI002  31/10/2022  Primavera   
129  Río Carapachay y Arroyo Gallo Fiambre     TI003  31/10/2022  Primavera   
130            Río Reconquista y Río Lujan     TI004  31/10/2022  Primavera   
131     Rio Tigre 100m antes del Rio Luján     TI005  31/10/2022  Primavera   
132         Río Lujan y Canal San Fernando     TI007  31/10/2022  Primavera   
133          Río Capitán y Río San Antonio     TI008  31/10/2022  Primavera   
134         Arroyo Abra Vieja y Santa Rosa     TI009  31/10/2022  Primavera   
135                               Del Arca     SF015  31/10/2022  Primavera   
137                      Reserva Ecológica     SI022  31/10/2022  Primavera   
138               Playa Espigón de Pacheco     SI024

Observamos que para la columna "dbo_mg_l" para la campaña Primavera siguen habiendo ocurrencias Nan.
la mediana devuelve NaN, significa que no hay suficientes datos válidos en la campaña "Primavera" para calcular una mediana, y eso explicaría por qué no podemos reemplazar los valores NaN. En ese caso, podríamos intentar un enfoque alternativo, como usar un valor por defecto o una mediana calculada de otra campaña.

In [164]:
# Calcular la mediana total de la columna 'dbo_mg_l' sin filtrar por campaña
median_total = copy_dataset['dbo_mg_l'].median()
# Reemplazar los valores NaN en 'dbo_mg_l' para la campaña 'Primavera' con la mediana total
copy_dataset.loc[(copy_dataset['campaña'] == 'Primavera') & (copy_dataset['dbo_mg_l'].isna()), 'dbo_mg_l'] = median_total


verificamos que se reemplazaron los Nan correctamente

In [165]:
copy_dataset["dbo_mg_l"]

0     6.200
1     5.800
2     1.900
3     5.800
4     2.600
       ... 
163   5.000
164   5.000
165   5.000
166   5.000
167   5.000
Name: dbo_mg_l, Length: 154, dtype: float64

Ahora si estamos en condiciones!

Realizamos un describe para ver una serie de caracteristicas descriptivas

Ya tenemos el dataset preparado para empezar con el modelado de variables y deducir relaciones entre distintas variables

In [166]:
pd.set_option('display.float_format', lambda x: '%.3f' % x)

seteamos la opcion para que el dataset muestre los valores completos y no en notacion cientifica

In [167]:
copy_dataset.describe()

tem_agua  tem_aire      od      ph  olores   color  espumas  mat_susp  \
count   154.000   154.000 154.000 154.000 154.000 154.000  154.000   154.000   
mean     17.900    15.826   6.777   7.439   0.078   0.084    0.032     0.175   
std       4.786     5.023   2.642   0.697   0.269   0.279    0.178     0.381   
min       6.000     4.000   0.360   5.000   0.000   0.000    0.000     0.000   
25%      14.933    13.000   5.363   7.000   0.000   0.000    0.000     0.000   
50%      17.850    14.000   6.805   7.355   0.000   0.000    0.000     0.000   
75%      20.775    19.000   8.170   7.790   0.000   0.000    0.000     0.000   
max      27.400    27.000  17.610  10.020   1.000   1.000    1.000     1.000   

       colif_fecales_ufc_100ml  escher_coli_ufc_100ml  ...  dbo_mg_l  \
count                  154.000                154.000  ...   154.000   
mean                 86140.292               4068.110  ...     5.875   
std                 380084.458              15012.075  ...     4.571   
min                     80.000                  1.000  ...     1.900   
25%                   1200.000                100.000  ...     3.625   
50%                   3900.000                330.000  ...     5.000   
75%                  37500.000               1675.000  ...     6.425   
max                4200000.000             150000.000  ...    42.000   

       dqo_mg_l  turbiedad_ntu  hidr_deriv_petr_ug_l  cr_total_mg_l  \
count   154.000        154.000               154.000        154.000   
mean     44.188         34.762                 0.101          0.544   
std      23.477         23.854                 0.008          1.946   
min      29.000          2.500                 0.100          0.005   
25%      30.000         18.050                 0.100          0.005   
50%      30.000         27.500                 0.100          0.005   
75%      52.250         45.000                 0.100          0.005   
max     180.000        130.000                 0.200         12.000   

       cd_total_mg_l  clorofila_a_ug_l  microcistina_ug_l     ica  \
count        154.000           154.000            154.000 154.000   
mean           0.001           263.861              0.189  44.071   
std            0.000           816.715              0.159   8.949   
min            0.001             0.000              0.150  23.000   
25%            0.001            10.000              0.150  38.000   
50%            0.001            10.000              0.150  42.000   
75%            0.001            41.800              0.200  50.000   
max            0.002          6410.000              1.670  76.000   

       calidad_de_agua  
count          154.000  
mean             2.584  
std              0.520  
min              1.000  
25%              2.000  
50%              3.000  
75%              3.000  
max              3.000  

[8 rows x 25 columns]

Podemos observar lo siguiente:
- OD = Vemos que se distribuye moderadamente en los 3 cuartiles, pero que el minimo es un valor muy bajo (podria ser un outlier o una zona influenciada por el Riachuelo donde la contaminacion es muy alta). Además vemos que el máximo valor registrado para esta columna es de 17, lo que sorprende un poco, a examinar.
- TEM_AGUA = el valor minimo es de 6 (°C) y el maximo de 27.4 (°C), el promedio de 17,89 (°C). Parece todo normal dentro de los cuartiles
- TEM_AIR = se comporta muy parecido a la temperatura del agua.
- OLORES, COLOR, ESPUMAS Y MAT_SUSP no aportan mucho ya que se se realizo un mapeo dicotomico {0, 1}. podemos analizar a simple vista que el promedio en todos los casos es menor al 17% por lo que indicaria una tendencia a la ausencia de estos indicadores.
- COLIFORMES_FECALES_UFC_100ML = Vemos que hay numeros extremadamente grandes para este indicador, lo que daria a entender que el agua esta muy contaminada, AUNQUE! el maximo valor es 4.200.000 en 100 ml lo que es demasiado grande para ser real. examinar.